### PR TITLE
GraphQL Query: Send Application

### LINKED ISSUE #308: [BE] Create send application GraphQL mutation
Create a GraphQL mutation that calls the following stored procedure:

* `OA_PKG_APP.SP_SEND_APPLICATION`

This should replicate and and all logic found on the existing legacy page:

* `/Forms/Application/Submit.aspx`

### LINKED ISSUE #309: [BE] Create confirmation details GraphQL query and email
After submitting an application, the legacy OneApp server redirects to a confirmation page at:

* `/Forms/Application/Confirmation.aspx`

This confirmation page calls the following stored procedure:

* `OA_PKG_GEN.SP_GET_CONFIRMATION_DETAILS`

The confirmation page also sends a confirmation email to the applicant, if they provided an email address.

While the legacy OneApp server handles this as a separate page (returning confirmation info, and sending an email), this may actually make more sense as follow up operations to the send application mutation (defined in #308). Before building these as separate GraphQL queries, we should see if it makes sense to integrate this as a response to the send application mutation.

As it may be possible to log back in after submitting an application and view confirmation information after the fact, we also may opt to do both (return the confirmation info via the send application mutation and stand it up as a separate GraphQL query).

### PR DESCRIPTION
**Pull request by ronaldheft-usds**
* Creates a new `applicationSend` mutation to support sending the application to UAP (Resolves #308).
  * This calls the stored procedure `SP_SEND_APPLICATION`.
  * `SP_SEND_APPLICATION` expects a `DISCLAIMER_UNDERSTOOD` value, which has been added as a GraphQL ENUM `ApplicationDisclaimerUnderstood`.
  * This mutation returns a new `APPLICATION_NUMBER` that the front-end must reference going forward. The previous `APPLICATION_NUMBER` that matches the `USER_ID` is no longer valid after submission.
* Creates a new `applicationConfirmation` query to return the confirmation details of the submitted application (Resolves #309).
  * This query requires the `APPLICATION_NUMBER` from `applicationSend`.
  * There is security in place to ensure a user can only access their submitted application (so they can't lookup other users' applications). _<- This could use some extra validation to ensure I implemented it correctly._
* This PR _does not_ handle sending the confirmation email to the user.
  * Upon further investigation, I discovered the current confirmation sent is literally a copy of the RAW rendered HTML from ASP.net (https://github.com/newjersey/dhs-oneapp-legacy/blob/master/NJOneAppAuto/Forms/Application/Confirmation.aspx.cs#L91).
  * This is not easy to replicate under the new API (since the API is not rendering HTML), and seems like a really bad user experience.
  * I'd recommend we just drop the confirmation email from the MVP scope at this time. If Protech would like to see an example of how to implement the confirmation email, we can design an email and use that instead of sending legacy HTML. _<- Thoughts @allyceh?_
* Updates docs to match all new queries / mutations.

### CODE REVIEW

File: oneapp-server/types/Application.js
Comment (mayaisrani) [inline oneapp-server/types/Application.js:55]: nit: County

File: oneapp-server/dao/ApplicationDao.js
Comment (mayaisrani) [inline oneapp-server/dao/ApplicationDao.js:50]: Any reason you're using this stored procedure instead of [OA_PKG_GEN's](https://github.com/newjersey/dhs-oneapp-legacy/blob/080a57380e9dedd834d2b6de141d7a031825159e/OneAppStoredProceduresFunctions/OA_PKG_GEN.sql#L452)? I can't find this call in OA_PKG_APP, and the legacy code seems to call OA_PKG_GEN?

File: oneapp-server/types/Application.test.js
Comment (mayaisrani) [inline oneapp-server/types/Application.test.js:3]: Let's rebase atop the new Application.test.js file introduced in #440

**Review by mayaisrani — CHANGES_REQUESTED** [review]
A few questions/nits inline

File: oneapp-server/types/Application.js
Comment (ronaldheft-usds) [reply oneapp-server/types/Application.js]: Good catch. Fixed!

File: oneapp-server/types/Application.test.js
Comment (ronaldheft-usds) [reply oneapp-server/types/Application.test.js]: Agreed. I want to hold this PR until all of yours are in, because I also need to do an end-to-end test against the whole application submission flow.

File: oneapp-server/dao/ApplicationDao.js
Comment (ronaldheft-usds) [reply oneapp-server/dao/ApplicationDao.js]: Ooo, good call. I had copied / pasted from send application and had the wrong package prefix. I also hadn't tested this stored procedure since the copy/paste.

I locally disabled the user protection (since I can't test `sendApplication` yet) and verified this stored procedure is now working and calling the correct package. I also made a few type changes on the output to make for a cleaner API.

**Comment by ronaldheft-usds** [comment]
Ready for review again. Planning on rebasing once all the existing PRs are merged in, because this we can test this end-to-end.

**Comment by ronaldheft-usds** [comment]
Merged in latest main. Running through a full end-to-end test.

File: oneapp-server/dao/ApplicationDao.js
Comment (mayaisrani) [inline oneapp-server/dao/ApplicationDao.js:59]: Nice. I think we need to comb through the other application component files, change the Y/N char types to booleans, and process the fields in the fetching functions. I can work on it if you're not already doing so.

File: oneapp-server/docs/config.yml
Comment (mayaisrani) [inline oneapp-server/docs/config.yml:97]: Thanks!

**Review by mayaisrani — APPROVED** [review]
Looks great and ready to merge!

File: oneapp-server/dao/ApplicationDao.js
Comment (ronaldheft-usds) [reply oneapp-server/dao/ApplicationDao.js:59]: I agree on that change if it's a straightforward boolean. I think we ran into the issue where some of the output was translated fields vs. a Y/N.

Maybe if the only value the stored procedure uses the language for is the translated value, we can always pass english into it (index 0), then check on the english translated text?

Comment (mayaisrani) [reply oneapp-server/dao/ApplicationDao.js:59]: Ah right. Maybe then I'll hold off and stand by until we're testing with @mikepif and the front end

**Pull request merged**
